### PR TITLE
Restrict work items to pr links

### PR DESCRIPTION
**Pull request by tdelabro**
closes E-169

### CODE REVIEW

**Review by AnthonyBuisset — APPROVED** [review]

File: backend/api/src/domain/payment/reason.rs [hunk 1]
```diff
@@ -1,7 +1,50 @@
...
+	#[error("A reason is not a valid url: {0}")]
+	NotAnUrl(String),
+	#[error("A reason is not a valid github url: {0}")]
+	NotGithubUrl(String),
+	#[error("A reason is not a valid github pull request url: {0}")]
+	NotPRUrl(String),
+}
+
+impl Reason {
+	pub fn is_valid(&self) -> Result<(), ValidReasonError> {
```
Comment (ofux) [inline backend/api/src/domain/payment/reason.rs:23]: I think a few unit tests would have been cool here to test all edge cases

**Review by ofux — APPROVED** [review]

File: backend/api/src/domain/payment/reason.rs (hunk 1 above)
Comment (tdelabro) [reply backend/api/src/domain/payment/reason.rs:24]: done
